velox/type/TimestampConversion.cpp

```diff
@@ -202,6 +206,10 @@ bool tryParseDateString(
       break;
     }
   }
+  // Spark digits of year must >= 4.
+  if (mode == ParseMode::kSparkCast && pos - sign < 4) {
```

Comment:
if we only use sign in arithmetics, should we declare it as an int (signCount or signPos)?

Also, what do we want to check for exactly? That the year part has at least four digits or that the year part of the string is at position 4? If the latter, what if you have leading whitespaces? 

In any case, could we make the comment a bit more explicit?

Reply:
Leading whitespaces have been skipped before, so bool type is enough. The year part has at least four digits, the following formats are allowed:
[+-]yyyy* [+-]yyyy*-[m]m [+-]yyyy*-[m]m-[d]d [+-]yyyy*-[m]m-[d]d [+-]yyyy*-[m]m-[d]d * [+-]yyyy*-[m]m-[d]dT*

Reply:
I assume white-spaces are removed at below location. Thanks.
https://github.com/facebookincubator/velox/blob/7e4d6263cac225e830369e5b4951be12040a2e43/velox/expression/CastExpr-inl.h#L285